References: bsc#1071825
Running under python3 the delete-event as defined in details.ui and connected
in details.py does not get called when the top right (x) button on the dialog
is clicked. It appears that even though this code is unchanged and works on python2,
it is not recognized as a top level dialog on python3 and therefore we explicitly
need to set the connection event. Not getting the event to cleanup leaves us in a
bad state for when the details dialog is reopened.

Index: virt-manager-4.1.0/virtManager/vmwindow.py
===================================================================
--- virt-manager-4.1.0.orig/virtManager/vmwindow.py
+++ virt-manager-4.1.0/virtManager/vmwindow.py
@@ -134,6 +134,9 @@ class vmmVMWindow(vmmGObjectUI):
             "on_details_menu_view_autoconnect_activate": self._autoconnect_ui_changed_cb,
         })
 
+        # Make sure the delete event is properly registered for the details dialog
+        self.topwin.connect("delete-event", self._window_delete_event)
+
         # Deliberately keep all this after signal connection
         self.vm.connect("state-changed", self._vm_state_changed_cb)
         self.vm.connect("resources-sampled", self._resources_sampled_cb)
